imap_processing/hit/l0/decom_hit.py

```diff
@@ -409,6 +385,53 @@ def assemble_science_frames(sci_dataset: xr.Dataset) -> xr.Dataset:
     return sci_dataset
 
 
+def decompress_rates_16_to_32(packed: int) -> int:
+    """
+    Will decompress rates data from 16 bits to 32 bits.
+
+    This function decompresses the rates data from the binary
+    format to integers. The compressed integer (packed) combines
+    two parts:
+
+    1. Mantissa: Represents the significant digits of the value.
+    2. Exponent: Determines how much to scale the mantissa (using powers of 2).
+
+    These parts are packed together into a single 16-bit integer.
+    Numbers up to 212 are uncompressed.
+
+    Parameters
+    ----------
+    packed : int
+        Compressed 16-bit integer.
+
+    Returns
+    -------
+    decompressed_int : int
+        Decompressed integer.
+    """
+    # In compressed formats, the exponent and mantissa are tightly packed together.
+    # The mask ensures you correctly separate the mantissa (useful for reconstructing
+    # the value) from the exponent (used for scaling).
+    # set to 16 bits
+    output_mask = 0xFFFF
+
+    # Packed is the compressed integer
+    # Right bit shift to get the exponent
+    power = packed >> MANTISSA_BITS
```

Comment:
Based on this, it looks power will only be greater than 1 if packed is >=8192. From the algorithm if the power is less than 1 then the decompressed number = the compressed number. So that makes sense with what you're seeing in the sample data, despite what the algorithm document seems to say